servers/src/test/java/tachyon/worker/block/evictor/EvictorTest.java

```diff
@@ -0,0 +1,208 @@
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Files;
+
+import tachyon.Constants;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+
+/**
+ * This class tests general behavior of all types of {@link Evictor} defined in {@link EvictorType}.
+ */
+@RunWith(Parameterized.class)
+public class EvictorTest {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static final int USER_ID = 2;
+  private static final long BLOCK_ID = 10;
+  private BlockMetadataManager mMeta;
+  private EvictorType mEvictorType;
+  private Evictor mEvictor;
+  private File mTempFolder;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    // Run this test against all types of Evictors
+    List<Object[]> list = new ArrayList<Object[]>();
+    for (EvictorType type : EvictorType.values()) {
+      if (type != EvictorType.DEFAULT) {
+        list.add(new Object[] {type});
+      }
+    }
+    return list;
+  }
+
+  public EvictorTest(EvictorType evictorType) {
+    mEvictorType = evictorType;
+  }
+
+  @Before
+  public final void before() throws IOException {
+    mTempFolder = Files.createTempDir();
+    mMeta = EvictorTestUtils.defaultMetadataManager(mTempFolder.getAbsolutePath());
+    mEvictor = EvictorFactory.create(mEvictorType, mMeta);
+  }
+
+  @After
+  public final void after() throws IOException {
+    FileUtils.forceDelete(mTempFolder);
+  }
+
+  @Test
+  public void noNeedToEvictTest1() throws IOException {
+    // metadata manager is just created, no cached block in Evictor
+    for (StorageTier tier : mMeta.getTiers()) {
+      for (StorageDir dir : tier.getStorageDirs()) {
+        Assert.assertTrue(mEvictor.freeSpace(dir.getCapacityBytes(),
+            EvictorTestUtils.location(tier, dir)).isEmpty());
+      }
+    }
+  }
+
+  @Test
+  public void noNeedToEvictTest2() throws IOException {
+    // cache some data in a dir, then request the remaining space from the dir
+    StorageDir dir = EvictorTestUtils.randomDir(mMeta);
+    long capacity = dir.getCapacityBytes();
+    long cachedBytes = capacity / 2 + 1;
+    EvictorTestUtils.cache(USER_ID, BLOCK_ID, cachedBytes, dir, mMeta, mEvictor);
+    Assert.assertTrue(mEvictor.freeSpace(capacity - cachedBytes, EvictorTestUtils.location(dir))
+        .isEmpty());
+  }
+
+  @Test
+  public void noNeedToEvictTest3() throws IOException {
+    // fill in all dirs except for a random directory in a random tier, then request the capacity of
+    // the directory with anyDirInTier
+    StorageDir dirLeft = EvictorTestUtils.randomDir(mMeta);
```

Comment:
In unit test, it is desired to make everything deterministic and avoid having randomness, so failures could be repeat easily. You can simply pick one dir according to the default manager